general/contributing/sourcetree.md

```diff
@@ -0,0 +1,48 @@
+---
+uid: contrib-source
+title: Source Tree
+---
+
+# Contributing to Jellyfin's Source Tree
+
+The Jellyfin Source Tree is a maze of many projects. With Emby's old code and Jellyfin's new code intermixed together. To help with this, this Source Tree guide was created.
+
+### [Jellyfin Server](https://github.com/jellyfin/jellyfin)
+1.  BDInfo: `Blu-Ray Analyzer`
+    - Properties: `Assembly Info`
+2.  DvdLib: `DVD Anaylzer`
+    - Ifo: 
+    - Properties:
+3.  Emby.Dlna: 
+4.  Emby.Drawing:
+5.  Emby.Naming:
+6.  Emby.Notifications:
+7.  Emby.Photos:
+8.  Emby.Server.Implementations:
+9.  Jellyfin.Api:
+10. Jellyfin.Drawing.Skia:
+11. Jellyfin.Server:
+12. MediaBrowser.Api:
+13. MediaBrowser.Common:
+14. MediaBrowser.Controller:
+15. MediaBrowser.LocalMetadata:
+16. MediaBrowser.MediaEncoding:
+17. MediaBrowser.Model:
+18. MediaBrowser.Providers:
+19. MediaBrowser.WebDashboard:
+20. MediaBrowser.XbmcMetadata:
+21. RSSDP:
+22. benches/Jellyfin.Common.Benches:
+
+
+### [Jellyfin WebOS](https://github.com/jellyfin/jellyfin-web)
+1.  Src: 
+    - components:
+    - controllers:
+    - css:
+    - img:
+    - legacy:
+    - libraries:
+    - scripts:
+    - splashscreens:
+    - strings:
```

Comment:
These folders have already changed a bit in some of my new pull requests (all static assets moved to a single folder) but I'll list the unchanged ones below.

* components: custom elements used for different sections of the user interface
* controllers: scripts that handle the logic for different pages
* legacy: currently used for all polyfills and scripts related to backwards compatibility
* libraries: dependencies that we eventually want to remove and include during the build step
* scripts: any script that isn't tied to a UI element or page but rather general functionality
* strings: translations for the entire interface

Some of the new changes I'm planning at the moment are below.

* assets: images, styles, splash screens, and any other static assets
* elements: custom UI components that are used globally such as buttons or menus
* themes: all themes will exist in this folder, including custom themes at some point

We also have a basic plugin system that could also get its own folder in the future, but it would need a lot of cleanup before that can happen. The HTML files for the pages should also be moved to a folder at some point (maybe added next to the controllers? not sure) so they can stay organized as well.

Reply:
thanks :) I'll add in descriptions for everything